ccvm_simulators/problem_classes/boxqp/problem_instance.py

```diff
@@ -143,10 +157,18 @@ def load_instance(
                 for idx in range(0, problem_size):
                     rval_v[idx] = -torch.Tensor([float(line_data_v[idx])])
                 # Read in the q_matrix matrix
-                for idx, line in enumerate(lines[2:]):
+                for idx, line in enumerate(lines[2 : problem_size + 2]):
                     line_data = line.split("\n")[0].split(file_delimiter)
                     for j, value in enumerate(line_data[:problem_size]):
                         rval_q[idx, j] = -torch.Tensor([float(value)])
+
+                # Read the last line as an additional information
+                last_raw_data_line = lines[problem_size + 2].split("\n")[0].split(file_delimiter)
+                solution_vector = []
+                for v in last_raw_data_line:
+                    if not v == "":
+                        solution_vector.append(float(v))
+                        
```

Comment:
Is the solution vector is required before parsing? If it is optional we will need to wrap this in a try block to allow the problem to be read without error in the case that the solution vector is not given:
```suggestion
                 # Read the last line as an additional information
                solution_vector = []
                try:
                    last_raw_data_line = lines[problem_size + 2].split("\n")[0].split(file_delimiter)
                    for v in last_raw_data_line:
                        if not v == "":
                            solution_vector.append(float(v))
                except IndexError:
                    # solution_vector was not supplied
                    pass
```